src/kaleidoscope/HardwareDescription.h

```diff
@@ -0,0 +1,33 @@
+/* -*- mode: c++ -*-
+ * HardwareDescription -- Hardware description base class
+ * Copyright (C) 2019  Keyboard.io, Inc
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "kaleidoscope/driver/scanner/NoOpScanner.h"
+#include "kaleidoscope/driver/led/NoLeds.h"
+#include "kaleidoscope/driver/mcu/NoMCU.h"
+
+namespace kaleidoscope {
+
+struct HardwareDescription {
+  typedef kaleidoscope::driver::scanner::NoOpScannerDescription ScannerDescription;
+  typedef kaleidoscope::driver::scanner::NoOpScanner<ScannerDescription> KeyScanner;
```

Comment:
I went with `Scanner`, because I was anticipating other kinds of scanners (trackballs, thumbsticks, that kind of stuff). But then those won't be using a matrix most likely, so won't derive from this base struct, so your suggestion is valid. Thanks!